src/Core/Name.idr

```diff
@@ -75,6 +75,16 @@ isUserName (NS _ n) = isUserName n
 isUserName (DN _ n) = isUserName n
 isUserName _ = True
 
+export
+||| Test whether a user name begins with an operator symbol.
+isOpName : Name -> Bool
+isOpName n = let Just n = userNameRoot n
+                 | Nothing => False
+             in let n' = unpack n
+             in case n' of
+                  []     => False
+                  c :: _ => c `elem` (unpack ":!#$%&*+./<=>?@\\^|-~")
```

Comment:
Hm, if for some reason this string containing valid operator symbols is changed in the lexer, it will be harder to debug why printing is broken again :)
Should we decouple it and place somewhere accessible to both the lexer and the pretty-printer ?

Reply:
Thank you for reviewing! I think that would be nice to unify them. I will commit some changes but still not sure where is the suitable place. 0.0

Reply:
That works